client/parties/party-view.tsx

```diff
@@ -129,35 +130,29 @@ function renderPartyMessage(msg: Message) {
   }
 }
 
-interface PartyViewProps {
-  params: { partyId: string }
-}
-
-export function PartyView(props: PartyViewProps) {
+export function PartyView() {
   const dispatch = useAppDispatch()
-  const partyId = decodeURIComponent(props.params.partyId).toLowerCase()
-  const prevPartyId = usePrevious(partyId)
   const party = useAppSelector(s => s.party)
-  const isInParty = !!party.id
-  const prevIsInParty = usePrevious(isInParty)
-  const isLeavingParty = prevIsInParty && !isInParty && prevPartyId === partyId
+  const partyId = party.id
+  const prevPartyId = usePrevious(partyId)
+  // Party can change when a user accepts an invite to a new party while already being a member of a
+  // different party that is currently being rendered.
+  const partyChanged = !!prevPartyId && prevPartyId !== partyId
 
-  // TODO(2Pac): Pull this out into some kind of "isLeaving" hook and share with chat/whispers/lobby
   useEffect(() => {
-    if (isLeavingParty) {
-      push('/')
-    }
-  }, [isLeavingParty])
+    const isInParty = !!partyId
 
-  useEffect(() => {
     if (isInParty) {
+      if (partyChanged) {
+        push(urlPath`/parties/${partyId}`)
```

Comment:
`push` doesn't make sense here, the previous URL is effectively a location that no longer exists for this user, so why are we keeping it in the back stack?